overlord/snapshotstate/backend/reader.go

```diff
@@ -268,10 +268,15 @@ func (r *Reader) Restore(ctx context.Context, usernames []string, logf Logf) (rs
 			return rs, fmt.Errorf("Cannot restore snapshot into %q: not a directory.", parent)
 		}
 
+		// TODO: have something more atomic in osutil
 		tempdir, err := ioutil.TempDir(parent, ".snapshot")
 		if err != nil {
 			return rs, err
 		}
+		if err := sys.ChownPath(tempdir, uid, gid); err != nil {
```

Comment:
Do you need to chmod it as well (to something sane?) or do you control that elsewhere?